### PR TITLE
Add pre-commit and linting

### PR DESCRIPTION
**Pull request by zerolab**
This PR:

- adds configuration for black/flake8/isort and [pre-commit](https://pre-commit.com/) support
- does one-pass lint
- adds linting to CI
- updates badges in README

### CODE REVIEW

**Comment by dfunckt** [comment]
Great stuff @zerolab! Can you please split the changes related to dropping deprecated code into another PR?

**Comment by zerolab** [comment]
> Can you please split the changes related to dropping deprecated code into another PR?

Done

File: tests/testsuite/test_predicates.py [hunk 1]
```diff
@@ -251,39 +267,42 @@ def test_one_arg(self):
...
         assert p.num_args == 2
-        p.test('a', 'b')
+        p.test("a", "b")
 
     def test_no_mask(self):
         @predicate
         def p(a=None, b=None, *args, **kwargs):
             assert len(args) == 0
             assert len(kwargs) == 1
-            'c' in kwargs
```
Comment (dfunckt) [inline tests/testsuite/test_predicates.py:273]: What was I thinking here? :)

**Review by dfunckt — APPROVED** [review]
This is brilliant, thank you.

File: tests/testsuite/test_predicates.py (hunk 1 above)
Comment (zerolab) [reply tests/testsuite/test_predicates.py:273]: would not have spotted it for a million years. flake8 FTW :)

**Pull request merged**